### PR TITLE
*: integrate nflog

### PR DESCRIPTION
**Pull request by fabxc**
This commit replaces the previous NotifyInfo provider with the new
nflog package. It needs adjustments in the behavior of the deduping
stage.
The nflog stores notification digests per receiver per alert aggregation
group rather than one entry for alert per receiver. This drastically
reduces the number of entries and removes interference
across aggregation groups.

@brancz

### CODE REVIEW

File: notify/notify.go [hunk 1]
```diff
@@ -384,74 +401,127 @@ func (fr *FilterResolvedStage) Exec(ctx context.Context, alerts ...*types.Alert)
...
-			// the repeat interval has already passed.
-			if !now.After(last.Timestamp.Add(interval)) {
-				return false
-			}
+func utcNow() time.Time {
+	return time.Now().UTC()
+}
+
+// TODO(fabxc): this could get slow, but is fine for now. We may want to
+// have something mor sophisticated at some point.
```
Comment (brancz) [inline notify/notify.go]: more

File: notify/notify_test.go [hunk 2]
```diff
@@ -235,86 +281,7 @@ func TestRoutingStage(t *testing.T) {
 }
 
 func TestSetNotifiesStage(t *testing.T) {
```
Comment (brancz) [inline notify/notify_test.go]: missing test?

Comment (fabxc) [inline notify/notify_test.go]: Ah, that was not intended.

Comment (fabxc) [inline notify/notify_test.go]: PTAL

**Comment by brancz** [comment]
LGTM 👍

**Pull request merged**